src/main.rs

```diff
@@ -50,10 +50,64 @@ fn main() {
     }
 }
 
+fn build_cache_without_training(cl: clap::ArgMatches) -> Result<(), Box<dyn Error>> {
```

Comment:
Consider adding a docstring-like comment describing what this function does (e.g., as in https://github.com/outbrain/fwumious_wabbit/blob/fe4a42fb4da27fc5a913e86210be2a5b13d2487d/src/model_instance.rs#L358)

Reply:
done